### PR TITLE
Gender Cleanup

### PR DESCRIPTION
**Pull request by bretjb**
## What does this do?

- Bumps typescript + babel up minor versions to latest
- Removes shadowed `_gender` variable, preferring just the base `gender` on NPCs
- Switches the order of linting to let typescript linter go last, removing some incorrectly parsed js linting rules on ts (e.g., `key in foo`)
- Cleaned up genderData and tested it

## How was this tested? Did you test the changes in the compiled `.html` file?

Unit tests + poking around compiled html

## Is there a [GitHub Issue](https://github.com/ryceg/Eigengrau-s-Essential-Establishment-Generator/issues?q=is%3Aopen+is%3Aissue) that this is resolving?

Nope!

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
-  "extends": ["plugin:@typescript-eslint/recommended", "standard"],
+  "extends": [
+    "standard",
+    "plugin:@typescript-eslint/recommended"
```
Comment (Glinkis) [inline .eslintrc:4]: We could  perhaps add `overrides: { files: ["**/*.ts"] }` to only run the typescript rules for the relevent files? Not sure exactly what the difference would be.

File: lib/src/utils.ts [hunk 2]
```diff
@@ -26,6 +26,16 @@ export function assign<T, S> (target: T, source: S): asserts target is T & S {
   Object.assign(target, source)
 }
 
+type Key = string | number | symbol;
+
+export function has<K extends Key, T> (key: K, t: T): t is T & Record<K, unknown> {
+  return key in t
+}
+
+export function isObject (maybeObj: unknown): maybeObj is Record<string | number | symbol, unknown> {
```
Comment (Glinkis) [inline lib/src/utils.ts:35]:
```suggestion
export function isObject (maybeObj: unknown): maybeObj is Record<Key, unknown> {
```

**Review by Glinkis — APPROVED** [review]
Looks great!

File: .eslintrc (hunk 1 above)
Comment (bretjb) [reply .eslintrc:4]: Oh, I think we actually want the typescript version of standard: https://github.com/standard/eslint-config-standard-with-typescript . Lemme see if that's a disruptive change.

Comment (Glinkis) [reply .eslintrc:4]: Haha yeah... Maybe as a separate migration.

**Pull request merged**